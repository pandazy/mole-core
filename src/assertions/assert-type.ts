import { assertValue } from './assert-value';
import { isA } from '../type-check/is-a';

export const assertType = (constructor: any) => (context: string) => (
	value: any,
) =>
	assertValue({
		isValid: isA(constructor),
		failureCause: `type must be ${constructor.name}`,
	})(context)(value);
