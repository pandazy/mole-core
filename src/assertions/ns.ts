import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
