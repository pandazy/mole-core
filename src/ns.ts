export function ns(context: string): string {
  return `Mole:${context}`;
}
