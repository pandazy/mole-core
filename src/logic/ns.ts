import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
