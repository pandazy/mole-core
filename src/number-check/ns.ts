import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
