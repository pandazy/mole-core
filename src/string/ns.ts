import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
