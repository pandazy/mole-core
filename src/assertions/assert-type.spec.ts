import { assertType } from './assert-type';
import { ns } from './ns';

describe(ns('assertType'), () => {
  it('should assert a specific type', () => {
    expect.assertions(1);

    const assertArray = assertType(Array)('test@arg');

    expect(() => assertArray('a')).toThrowErrorMatchingSnapshot();
    assertArray([]);
  });
});
