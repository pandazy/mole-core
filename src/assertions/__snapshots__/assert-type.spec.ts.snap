// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:assertions:assertType should assert a specific type 1`] = `
"
[context]: test@arg
[failure cause]: type must be Array
[actual value]: \\"a\\"
[actual value type]: String"
`;
